source/backend/api/Controllers/LookupController.cs

```diff
@@ -76,102 +83,112 @@ public IActionResult GetPropertyClassificationTypes()
-            var activityTemplateTypes = _mapper.Map<Model.LookupModel[]>(_lookupRepository.GetActivityTemplateTypes());
-            var activityStatusTypes = _mapper.Map<Model.LookupModel[]>(_lookupRepository.GetActivityStatusTypes());
-            var acqFilePersonProfileTypes = _mapper.Map<Model.LookupModel[]>(_lookupRepository.GetAcqFilePersonProfileTypes());
-            var tenantTypes = _mapper.Map<Model.LookupModel[]>(_lookupRepository.GetTenantTypes());
-            var acqFundingTypes = _mapper.Map<Model.LookupModel[]>(_lookupRepository.GetAcquisitionFundingTypes());
 
-            var codes = new List<object>();
-            codes.AddRange(areaUnitTypes);
-            codes.AddRange(classificationTypes);
-            codes.AddRange(contactMethodTypes);
-            codes.AddRange(countries);
-            codes.AddRange(districts);
-            codes.AddRange(insuranceTypes);
-            codes.AddRange(leaseCategoryTypes);
-            codes.AddRange(leaseInitiatorTypes);
-            codes.AddRange(leasePaymentFrequencyTypes);
-            codes.AddRange(leasePaymentMethodTypes);
-            codes.AddRange(leasePaymentReceivableTypes);
-            codes.AddRange(leasePaymentStatusTypes);
-            codes.AddRange(leaseProgramTypes);
-            codes.AddRange(leasePurposeTypes);
-            codes.AddRange(leaseResponsibilityTypes);
-            codes.AddRange(leaseStatusTypes);
-            codes.AddRange(leaseTermStatusTypes);
-            codes.AddRange(leaseTypes);
-            codes.AddRange(organizationTypes);
-            codes.AddRange(propertyImprovementTypes);
-            codes.AddRange(propertyTypes);
-            codes.AddRange(provinces);
-            codes.AddRange(regions);
-            codes.AddRange(roleCodes);
-            codes.AddRange(securityDepositTypes);
-            codes.AddRange(tenureTypes);
-            codes.AddRange(researchStatusTypes);
-            codes.AddRange(requestSourceTypes);
-            codes.AddRange(researchPurposeTypes);
-            codes.AddRange(propertyResearchPurposeTypes);
-            codes.AddRange(propertyAnomalyTypes);
-            codes.AddRange(propertyRoadTypes);
-            codes.AddRange(propertyAdjacentLandTypes);
-            codes.AddRange(volumeUnitTypes);
-            codes.AddRange(propertyVolumetricTypes);
-            codes.AddRange(pphStatusType);
-            codes.AddRange(documentStatusTypes);
-            codes.AddRange(acquisitionFileStatusTypes);
-            codes.AddRange(acquisitionPhysFileStatusTypes);
-            codes.AddRange(acquisitionTypes);
-            codes.AddRange(activityTemplateTypes);
-            codes.AddRange(activityStatusTypes);
-            codes.AddRange(acqFilePersonProfileTypes);
-            codes.AddRange(tenantTypes);
-            codes.AddRange(acqFundingTypes);
+            if (!_memoryCache.TryGetValue(CacheKeys.Lookup, out JsonResult cachedLookupResponse))
```

Comment:
Store this result in memory, this should make all future requests for the lookup codes very fast and reduce need for db transactions until the server is restarted.